zulipterminal/ui_tools/views.py

```diff
@@ -1646,12 +1665,68 @@ def create_link_buttons(
 
         return link_widgets, link_width
 
+    def create_code_snippet_buttons(
+        self, controller: Any, code_snippets: List[Tuple[str, List[Tuple[str, str]]]]
+    ) -> Tuple[List[Any], int]:
+        code_snippet_widgets = []
+        code_snippet_width = 0
+
+        for index, snippet in enumerate(code_snippets):
+            language, snippet_list = snippet
+            language = "" if language is None else language
+            display_code, copy_code = self.get_code_from_snippet(snippet_list)
+            if display_code:
+                code_snippet_width = max(
+                    code_snippet_width, len(max(display_code, key=len))
+                )
+                # if language == "":
+                #     display_code[-1] = list(display_code[-1])
+                #     display_code[-1][-1] = (
+                #         display_code[-1][-1][:-1]
+                #         if display_code[-1][-1][-1] == "\n"
+                #         else display_code
+                #     )
+                #     display_code[-1] = tuple(display_code[-1])
+            code = f"{str(index+1)}: {language}\n"
+            button = CodeSnippetButton([("pygments:w", code)] + display_code)
+            urwid.connect_signal(button, "click", self.copy_to_clipboard, copy_code)
+            display_attr = None if index % 2 else "popup_contrast"
```

Comment:
It would be cleaner if we could have this kind of behavior, and that in a few other methods below, defined in the CodeSnippetButton itself instead, like we do with other buttons - I can't see a reason not to do this.

Note how the create_link_button function is simpler compared to this one.

The CodeSnippetButton does very little that's unique right now - it's very close to being a very simple Button.